syntax = "proto3";

package transport_catalogue_serialize;

import "map_renderer.proto";
import "transport_router.proto";

message Coordinates {
    double lat = 1;
    double lng = 2;
}

message Stop {
    bytes name = 1;
    Coordinates coordinates = 2;
}

message Bus {
    bool route_type = 1;
    bytes number = 2;
    repeated bytes stops = 3;
    uint32 route_stops_count = 4;
    uint32 route_length = 5;
    double curvature = 6;
    bytes final_stop = 7;
}

message Distance {
    bytes from = 1;
    bytes to = 2;
    uint64 distance = 3;
}

message TransportCatalogue {
    repeated Stop stops = 1;
    repeated Distance stops_distance = 2;
    repeated Bus buses = 3;
}

message DataBase {
    TransportCatalogue transport_catalogue = 1;
    renderer_serialize.MapRenderer map_renderer = 2;
    router_serialize.Router router = 3;
}